crates/util/cometindex/src/indexer.rs

```diff
@@ -188,8 +188,15 @@ impl Indexer {
             }
             // Mark that we got to at least this event
             update_watermark(&mut dbtx, event.local_rowid).await?;
-            dbtx.commit().await?;
+            // Only commit in batches of <= 1000 events, for about a 5x performance increase when
+            // catching up.
+            if scanned_events % 1000 == 0 {
```

Comment:
it seems possible to miss this if the while loop short circuits from the check at line 172, resulting in more than 1000 events being committed at once

Reply:
Switched to using the relevant_events as the metric, so that you commit every time you've processed 1000 relevant events, avoiding this issue.